src/contexts/ActivityContext.js

```diff
@@ -0,0 +1,248 @@
+}
+
+const TEN_MINUTES = 1000 * 60 * 10
+
+let storedList
+
+// Provides easy access to the user activities list
+class ActivityProvider extends React.Component {
+  static propTypes = {
+    account: EthereumAddressType, // Current wallet
+    children: PropTypes.node,
+    daoDomain: PropTypes.string, // domain of current DAO
+    walletWeb3: PropTypes.object,
+  }
+  state = {
+    // activities of all accounts
+    activities: [],
+  }
+  static defaultProps = {
+    account: '',
+  }
+
+  componentDidUpdate(prevProps) {
+    const { daoDomain, account } = this.props
+
+    if (daoDomain !== prevProps.daoDomain || account !== prevProps.account) {
+      const storageKey = `activity:${network.type}:${daoDomain}:${account}`
+      storedList = new StoredList(storageKey)
+
+      // eslint-disable-next-line react/no-did-update-set-state
+      this.setState(
+        { activities: storedList.getItems() },
+        this.refreshPendingActivities
+      )
+    }
+  }
+
+  componentWillMount() {
+    this._checkInterval = setInterval(this.checkForTimedOut, 1000 * 30)
+  }
+  componentWillUnmount() {
+    clearInterval(this._checkInterval)
+  }
+
+  // Refresh the status of pending activities
+  refreshPendingActivities() {
+    const { walletWeb3 } = this.props
+
+    this.state.activities
+      .filter(({ status }) => status === activityStatusTypes.PENDING)
+      .forEach(({ transactionHash }) => {
+        walletWeb3.eth
+          .getTransaction(transactionHash)
+          .then(({ blockNumber }) =>
+            blockNumber ? this.setActivityConfirmed(transactionHash) : null
+          )
+          .catch(console.error)
+      })
+  }
+
+  checkForTimedOut = () => {
+    const now = Date.now()
+    this.state.activities.forEach(activity => {
+      const timeDelta = now - activity.createdAt
+      if (
+        timeDelta > TEN_MINUTES &&
+        activity.status === activityStatusTypes.PENDING
+      ) {
+        // Set pending items to timed out after 10 minutes
+        this.setActivityTimedOut(activity.transactionHash)
+      }
+    })
+  }
+
+  add = activity => {
+    this.setState({
+      activities: storedList.add(activity),
+    })
+  }
+
+  addTransactionActivity = ({
+    transactionHash = '',
+    from = '',
+    targetAppProxyAddress = '',
+    forwarder = '',
+    forwarderProxyAddress = '',
+    description = '',
+  } = {}) => {
+    const newActivity = {
+      createdAt: Date.now(),
+      type: activityTypes.TRANSACTION,
+      status: activityStatusTypes.PENDING,
+      read: false,
+      transactionHash,
+      // account address from which the transaction was created
+      from,
+      targetAppProxyAddress,
+      forwarderProxyAddress,
+      forwarder,
+      description,
+    }
+
+    const updatedActivities = storedList.add(newActivity)
+
+    this.setState({ activities: updatedActivities })
+  }
+
+  currentAccountPredicate = ({ from }) =>
+    from.toLowerCase() === this.props.account.toLowerCase()
```

Comment:
We can use the [`addressesEqual()`](https://github.com/aragon/aragon/blob/master/src/web3-utils.js#L19) util instead :)